api/pkg/cli/cmd/check_upgrade/check_upgrade_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package check_upgrade
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	res "github.com/tektoncd/hub/api/gen/resource"
+	"github.com/tektoncd/hub/api/pkg/cli/test"
+	cb "github.com/tektoncd/hub/api/pkg/cli/test/builder"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	pipelinev1beta1test "github.com/tektoncd/pipeline/test"
+	"gopkg.in/h2non/gock.v1"
+	"gotest.tools/v3/golden"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic/fake"
+)
+
+var resVersion = &res.ResourceData{
+	ID:   1,
+	Name: "foo",
+	Kind: "Task",
+	Catalog: &res.Catalog{
+		ID:   1,
+		Name: "tekton",
+		Type: "community",
+	},
+	Rating: 4.8,
+	LatestVersion: &res.ResourceVersionData{
+		ID:                  12,
+		Version:             "0.2",
+		Description:         "v0.1 Task to run foo",
+		DisplayName:         "foo-bar",
+		MinPipelinesVersion: "0.11",
+		RawURL:              "http://raw.github.url/foo/0.1/foo.yaml",
+		WebURL:              "http://web.github.com/foo/0.1/foo.yaml",
+		UpdatedAt:           "2020-01-01 12:00:00 +0000 UTC",
+	},
+	Tags: []*res.Tag{
+		{
+			ID:   3,
+			Name: "cli",
+		},
+	},
+	Versions: []*res.ResourceVersionData{
+		{
+			ID:      12,
+			Version: "0.2",
+		},
+	},
+}
+
+func TestUpdateAvailable(t *testing.T) {
+	cli := test.NewCLI()
```

Comment:
I think we can create two functions as most of the code for all the tests looks same in all the test except mocking the api and getting the `exisitng` task, wdyt ??

Reply:
probably in follow-up PR we can try fixing these as the whole CLI tests needs to be fixed.